src/action/macos/create_determinate_volume_service.rs

```diff
@@ -0,0 +1,235 @@
+use serde::{Deserialize, Serialize};
+use tracing::{span, Span};
+
+use std::path::{Path, PathBuf};
+use tokio::{
+    fs::{remove_file, OpenOptions},
+    io::AsyncWriteExt,
+    process::Command,
+};
+
+use crate::action::{
+    Action, ActionDescription, ActionError, ActionErrorKind, ActionTag, StatefulAction,
+};
+
+/** Create a plist for a `launchctl` service to mount the volume
+ */
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+#[serde(tag = "action_name", rename = "create_determinate_volume_service")]
+pub struct CreateDeterminateVolumeService {
+    pub(crate) path: PathBuf,
+    mount_service_label: String,
+    needs_bootout: bool,
+}
+
+impl CreateDeterminateVolumeService {
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn plan(
+        path: impl AsRef<Path>,
+        mount_service_label: impl Into<String>,
+    ) -> Result<StatefulAction<Self>, ActionError> {
+        let path = path.as_ref().to_path_buf();
+        let mount_service_label = mount_service_label.into();
+        let mut this = Self {
+            path,
+            mount_service_label,
+            needs_bootout: false,
+        };
+
+        // If the service is currently loaded or running, we need to unload it during execute (since we will then recreate it and reload it)
+        // This `launchctl` command may fail if the service isn't loaded
+        let mut check_loaded_command = Command::new("launchctl");
+        check_loaded_command.arg("print");
+        check_loaded_command.arg(format!("system/{}", this.mount_service_label));
+        tracing::trace!(
+            command = format!("{:?}", check_loaded_command.as_std()),
+            "Executing"
+        );
+        let check_loaded_output = check_loaded_command
+            .output()
```

Comment:
nit: I don't think this needs to be `output` since you only check the `status()` of it.